### ISSUE TITLE
Add UniqueConstraint for Graph.slug where Graph.source_identifier = None

### ISSUE THREAD

**Issue by jacobtylerwalls** [archesproject/arches#11565]
There are four ways to get JSON-LD data into arches:

1. the POST api
2. the PUT api
3. the `load_jsonld` management command
4. the json-ld bulk import module, which is a wrapper around no. 3

All four allow you to pass a graph by slug. In dev/8.0.x we currently assume the user means the original graph. But if your tile data has already migrated away to some future graph version, that assumption would leave you working against an outdated version. 

Should we assume when these four places are given a graph slug, that the caller wants to work against the latest graph?

cc/ @robgaston

***
Noticed while dev'ing archesproject/arches-lingo#103, which might create a fifth place to look up graphs by slug.

**Comment by apeters** [comment]
I would agree that, although, we can never know where the data originally came from, assuming the latest graph makes sense to me.

**Comment by jacobtylerwalls** [comment]
I misunderstood the number of graph rows sharing a slug: should only ever be 2 no matter how many versions there are. The versioning happens in json. A UniqueConstraint across `slug` and `source_identifier=None` could potentially clarify this or improve performance slightly.
